src/ScriptEngine/ScriptEngine.cs

```diff
@@ -27,40 +27,59 @@ public class ScriptEngine : BaseUnityPlugin
 
         ConfigEntry<bool> LoadOnStart { get; set; }
         ConfigEntry<KeyboardShortcut> ReloadKey { get; set; }
+        
+        // FS Watcher EDIT
+        ConfigEntry<bool> QuietMode { get; set; }
+        ConfigEntry<bool> EnableFileSystemWatcher { get; set; }
+        ConfigEntry<bool> IncludeSubdirectories { get; set; }
+        
+        private FileSystemWatcher fileSystemWatcher;
+        private bool shouldReload;
+        //
 
         void Awake()
         {
             LoadOnStart = Config.Bind("General", "LoadOnStart", false, new ConfigDescription("Load all plugins from the scripts folder when starting the application"));
             ReloadKey = Config.Bind("General", "ReloadKey", new KeyboardShortcut(KeyCode.F6), new ConfigDescription("Press this key to reload all the plugins from the scripts folder"));
+            QuietMode = Config.Bind("General", "QuietMode", false, new ConfigDescription("Suppress sending log messages to console except for the error ones."));
+            EnableFileSystemWatcher = Config.Bind("General", "EnableFileSystemWatcher", false, new ConfigDescription("Listens for changes to the scripts directory and reloads all plugins automatically."));
+            IncludeSubdirectories = Config.Bind("General", "IncludeSubdirectories", false, new ConfigDescription("Include also subdirectories under the scripts folder."));
 
             if (LoadOnStart.Value)
                 ReloadPlugins();
+
+            if (EnableFileSystemWatcher.Value)
+                StartFileSystemWatcher();
         }
 
         void Update()
         {
-            if (ReloadKey.Value.IsDown())
+            if (shouldReload || ReloadKey.Value.IsDown())
```

Comment:
This can rapid fire reload while copying multiple files, ideally there would be a timer before the update happens. shouldReload could be changed into a float and set to say 3 by the watcher, then inside update subtract Time.unscaledDeltaTime from it and reload once it passes 0.

Reply:
> This can rapid fire reload while copying multiple files

Indeed, solved.